### PR TITLE
Update master branch with latest refactoring

### PR DESCRIPTION
**Pull request by osorensen**

### CODE REVIEW

**Comment by osorensen** [comment]
I think the linter issues are false positives which will disappear once the propsed changes to linter.yml get onto the master branch.

**Comment by wleoncio** [comment]
> I think the linter issues are false positives which will disappear once the propsed changes to linter.yml get onto the master branch.

What I've also done in other packages is to have `testthat.R` simply cointain `testthat::test_check("BayesMallows")`. This works around the `library()` complaint (and apparently a library call to the package itself is unnecessary).

**Review by wleoncio — APPROVED** [review]

**Comment by osorensen** [comment]
> 

That's a good idea. Fixed in latest commit.

**Comment by wleoncio** [comment]
Added a small contribution, couldn't help myself 😅

File: codecov.yml [hunk 1]
```diff
@@ -12,3 +12,5 @@ coverage:
         target: auto
         threshold: 1%
         informational: true
+ignore:
+    - "inst\examples\sample_mallows_example.R"
```
Comment (wleoncio) [inline codecov.yml:16]: Looks like this didn't work. 😞

**Pull request merged**